### PR TITLE
Feature/iab core update

### PR DESCRIPTION
**Pull request by alextremp**
## Description

This need was originated when we checked that the automatic minor version upgrades for `"@iabtcf/core": "1"` where breaking our consent acceptance restrictions & automatic vendor updates because of some changes on how the consent was decoded from `"@iabtcf/core": "1.1.0"` onwards.

Mainly, now:
- vendors with a `deletedDate` in the vendor list will not be encoded and so, our check of "all vendors has consent to true?" is not matching them as they'll come with a `false`
- decoded `consent.vendor.consents` / `consent.vendor.legitimateInterests` does not have all vendor ids from the vendor list in each node, will only have vendors with purpose usages and vendors with legitimate interest on purposes, respectively, so again, "all vendors has consent to true?" check would fail.

Additionally, thinking on users that approve  all vendors to use legitimate interests but reject all them for purpose usages, "all vendors has consent to true?" would raise an invalid consent again, because we were checking both nodes to have all a `true` or `false` on both sides. This PR also fixes that, and if an user has rejected all vendor consents but approved all vendor legitimate interests, a valid consent would be raised with new vendor list vendors added with a rejected purpose consent but approved legitimate interest consent.

This last behavior may cause also in fixed `"@iabtcf/core": "1.0.1"` to raise invalid consent if the consent was saved with an old vendor list, p.ex:

- using this valid consent with all vendor consents and legitimate interests accepted, but generated with vendor list v36, would raise an invalid consent instead of updating vendors.
  - cookie value: `CO8QNisO8QNisCBAPAENAkCsAP_AAH_AAAiQF7NX2T5OI2vjK2ZdF5BEaYwfZxyigMgChgQIsS4NwIeFbBoGL2AgHBG4JCQAGBAEEACBAQIkHGBcCQAAgIgBiRCMQEmMCzNKBJJAggEbM0FACCVmHkHSmZCY7068O__TMC7yCTBUnAJEhRCAkixAiFMAEI4gAAGAJQQgAQIWGgAICNgUBHqAAAAgMAAIAAAAQQAAAgAEAACAiAAAAQEAIAIgEAAAARoCEAAEgQAQAkSAAABADQkAIIghAEIODAKOEAAA.YAAAAAAAAAAA`
  - can be decoded here: https://iabtcf.com/#/decode

This happens in fotocasa now with iab's core version set to `1.0.1` and using an old vendor list encoded cookie (also happens in the local tcf/ui demo):

image

And more:
- we were using "unknown" hardcoded cookie values in some tests, that we could not regenerate or validate with a concrete vendor list version as they were saved with a latest vendor list (iab online decoder has the same problem)

So:

  - an utility has been created to generate consents
    - check `IABConsentHelperTest`
    - can be used just chaning the `skip` clause for the `only` clause, to generate consents with locally saved vendor lists

  - All tests now only run with 2 versions of the vendor lists:
    - an "oldest" vendor list, set to v36 and saved locally in ES and EN languages
    - a "latest" vendor list, set to v44 and saved locally in ES and EN languages

  - All "existing" cookies used in tests are now centralized, have an explanative ID, and all them have been generated using the `IABConsentHelperTest`

  - now tests will not use hardcoded vendors or consents of a 2-3 ids subset of what a real consent would have, and instead, all test will validate that a consent is saved according to vendor list values, and loaded according to IAB consent decoder returned values, so test are testing real data.

## Solves ticket/s

* https://jira.scmspain.com/browse/PSP-3647

## Expected behavior

- a valid consent, no matter if it was saved with vendor list v36, can be updated with new vendor list in same policyVersion, generating (and saving as renewed) a new valid consent with latest vendor list with updated vendors, and so, **not raising the TCF UI to the user again**:

  - using old cookie with all vendor consents accepted:
`CO8QNisO8QNisCBAPAENAkCsAP_AAH_AAAiQF7NX2T5OI2vjK2ZdF5BEaYwfZxyigMgChgQIsS4NwIeFbBoGL2AgHBG4JCQAGBAEEACBAQIkHGBcCQAAgIgBiRCMQEmMCzNKBJJAggEbM0FACCVmHkHSmZCY7068O__TMC7yCTBUnAJEhRCAkixAiFMAEI4gAAGAJQQgAQIWGgAICNgUBHqAAAAgMAAIAAAAQQAAAgAEAACAiAAAAQEAIAIgEAAAARoCEAAEgQAQAkSAAABADQkAIIghAEIODAKOEAAA.YAAAAAAAAAAA`

renewed consent cookie:
`CO8Tfk6O8Tfk6CBABAENA-CsAP_AAH_AAAiQG-td_X_fb39j-_59_9t0eY1f9_7_v20zjgeds-8Nyd_X_L8X4mM7vB36pq4KuR4Eu3LBAQFlHOHcTQmw6IkVqTPsak2Mr7NKJ7PEinMbO2dYGHtfn9VTuZKY797s___z__-_____79f_r-3_3_vp9V-gboASYal8BFmJYwEk0aVQogQhXEh0AIAKKEYWiawgJXBTsrgI_QQMAEBqAjAiBBiCjFkEAAAAASURACAHggEQBEAgABACpAQgAI0AQWAEgYBAAKAaFgBFAEIEhBkcFRymBARItFBPJWAJRc7GGEIZRQAA.YAAAAAAAAAAA`

note that renewed consent was generated with vendor list v62:
image

- an old valid consent with edited vendors, would be marked as invalid for new vendor list, setting old existing vendor consents to the value they had, but new vendors consents will be consented to `false`, and TCF UI will be raised to the user:

  - using old cookie with some vendor consents accepted and other rejected:
`CO8TKsQO8TKsQCBAPAENAkCtX__AAH_AAAiQF6pFSCoKAklCCSIUEpAEIQgVIhQggEAAggQAkCQEgIKBJAgEKkAgCBCQICAAEBAEEAABAAIAFCAUAQAAgAgBCBAIQAiECiJCAIIAgAASEQEACAEiFAFCCRAQpQqUEqqSIF3kEmCpOASJCiEBJFiBEKYAIRxAAAMASghAAgQsNAAQEbAoCPUAAABAYAAQAAAAggAABAAIAAEBEAAAAgIAQARAIAAAAjQEIAAJAgAgBIkAAACAGhIAQRBCAIQcGAUcIAAA.YAAAAAAAAAAA`

  - now the TCF UI is raised with old vendor statuses and new vendors set to false

image

## Review steps

can be tested locally by using the `cookie.js` prepared cookie values and using the adevinta components demo:
`npm run dev tcf/ui -- --link-package=../boros-tcf`

## Further considerations

- this lets again the `iab-core` dependency to version `1` so new minor upgrades will be automatically received. It has been locally tested with latest version, 
```
  "_from": "@iabtcf/core@1",
  "_id": "@iabtcf/core@1.1.4",
```

I can set the dependency fixed to that version, but it seems to be working all fine now.

## Memetized description

mandatory

### CODE REVIEW

**Review by oscar-raig — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
     "extends": "./node_modules/@s-ui/lint/stylelint.config.js"
   },
   "dependencies": {
-    "@iabtcf/core": "1.0.1",
+    "@iabtcf/core": "1",
```
Comment (paulusrex) [inline package.json:73]: To be discussed with the team. Are we sure to enable the automatic update to any 1 version? It's not preferable to enable de dependandbot to make the autobump and check if any test was broken?

Comment (alextremp) [reply package.json:73]: For me it's OK to use dependantbot too

Comment (alextremp) [reply package.json]: @paulusrex added dependabot in this PR and fixed the iab's dependency

**Review by paulusrex — APPROVED** [review]

File: src/test/infrastructure/service/IABConsentHelperTest.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      gvlFactory,
+      allPurposeConsents: givenAcceptance,
+      allPurposeLegitimateInterests: givenAcceptance,
+      allSpecialFeatures: givenAcceptance,
+      allVendorConsents: givenAcceptance,
+      allVendorLegitimateInterests: givenAcceptance,
+      editedVendorConsents: true,
+      editedVendorLegitimateInterests: false
+    })
+    // console.log('iabGenerateConsent', result)
```
Comment (maryammdot) [inline src/test/infrastructure/service/IABConsentHelperTest.js:26]: do you want to keep this console.log commented?

**Review by maryammdot — APPROVED** [review]

File: src/test/infrastructure/service/IABConsentHelperTest.js (hunk 2 above)
Comment (alextremp) [reply src/test/infrastructure/service/IABConsentHelperTest.js:26]: It has been useful to me (to know that output vendors where not all the vendorlist vendors p.ex.) but I can remove

**Pull request merged**